internal/controllers/nmc_reconciler.go

```diff
@@ -578,6 +583,12 @@ func (h *nmcReconcilerHelperImpl) SyncStatus(ctx context.Context, nmcObj *kmmv1b
 				Terminated.
 				FinishedAt
 
+			node := &v1.Node{}
+			if err := h.client.Get(ctx, types.NamespacedName{Name: nmcObj.Name}, node); err != nil {
+				logger.Error(err, "Failed to get node object", "node", nmcObj.Name)
+			} else {
+				status.BootId = node.Status.NodeInfo.BootID
```

Comment:
Since `SyncStatus` is called before `ProcessModuleSpec` - if we update the status here then when we get to `ProcessModuleSpec` then https://github.com/kubernetes-sigs/kernel-module-management/pull/1061/files#diff-4abf41a7395eeba9459df6b2fe59c88e93e33e83dd683d40878c61651de912f6R378 will always be false. Am I missing something?

Reply:
@ybettan it will be false in the Reconcile loop , when we are handling the successful termination of the worker pod. But when the reconcile loop will be called due to the Node event(node was rebooted), the bootId in the status will be the old one (since the pod does not exists), and then the condition in the ProcessModuleSpec will be true